osu.Game/Beatmaps/Drawables/Cards/HoverHandlingContainer.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input.Events;
+
+namespace osu.Game.Beatmaps.Drawables.Cards
+{
+    public class HoverHandlingContainer : Container
+    {
+        public Func<HoverEvent, bool>? Hovered { get; set; }
+        public Action<HoverLostEvent>? Unhovered { get; set; }
+
+        protected override bool OnHover(HoverEvent e) => Hovered?.Invoke(e) ?? base.OnHover(e);
```

Comment:
Likewise, if this is inlined it can always return `false` and the event exposed can be simplified.

Reply:
I need to be able to return `true` here for the expanded part to block clicks correctly.

Reply:
For symmetry with `OnHoverLost`, I've made `OnHover` always call base but then prefer the return value given by `Hovered` (if present). I think that makes sense?